conf/report/report-template.html

```diff
@@ -65,13 +65,80 @@
       $('#filter_wrapper').remove();
       iter = 1;
       var div = $('<div id="filter_wrapper"><button id="mainButton" onclick="addOptions()">Add</button></div>').insertAfter('#headline');
-      filter.length = 0;
+      filters.length = 0;
     }
 
     function applyFilter() {
       apply_active = true;
-      if(!document.getElementById('removeButton'));
+      if(!document.getElementById('removeButton'))
         var remove_all = $('<button id="removeButton" onclick="removeAll()">Remove all filters</button>').insertAfter('#applyButton');
+      
+      report = $('#report_table').DataTable();
+      var num_filters = filters.length;
+      var coverage, coverage_operator, type, type_operator, tool, tool_operator;
+      for(var i=0; i<num_filters; i++){
+        if(filters[i].mode === 'coverage'){
+          coverage_operator = filters[i].condition;
+          coverage = filters[i].value;
+        }
+        if(filters[i].mode === 'type'){
+          type_operator = filters[i].condition;
+          type = filters[i].value;
+        }
+        if(filters[i].mode === 'tool'){
+          tool_operator = filters[i].condition;
+          tool = filters[i].value;
+        }
+      }
+
+      report.columns().every(function() {
+        var column = this;
+        if(column.index() !== 0 && column.index() !== 1) {
+          var theadname = column.header().textContent.trim();
+
+          if(tool_operator === 'is'){
+            if(theadname === tool){
+              var array = [];
```

Comment:
Currently, the filter only works for "is" condition because the array is declared in inside the `report.columns().every()` and will cannot store the values from the previous columns. When I initialized the array outside the function, it did not have access to it and gave an error at `array.push()`

Reply:
You probably had access to the array initialized outside, but you're replacing it with a string in line 136 (`array = array.join('|');`)